module/VuFind/src/VuFind/Db/Service/UserService.php

```diff
@@ -128,6 +128,54 @@ public function getUserByField(string $fieldName, int|string|null $fieldValue):
         throw new \InvalidArgumentException('Field name must be id, username, email or cat_id');
     }
 
+    /**
+     * Retrieve a user object by catalog ID. Returns null if no match is found.
+     *
+     * @param string $catId Catalog ID
+     *
+     * @return ?UserEntityInterface
+     */
+    public function getUserByCatId(string $catId): ?UserEntityInterface
+    {
+        return $this->getUserByField('cat_id', $catId);
```

Comment:
I realize it would be more efficient to proxy the database table methods here and then call these from getUserByField rather than the reverse -- but doing it this way will be more forward-compatible with #2233, so I think it's worth the trivial performance hit to keep the code more forward-compatible and reduce diffs.